packages/redux/src/analytics/middlewares/__tests__/bag.test.ts

```diff
@@ -317,98 +333,117 @@ describe('bagMiddleware()', () => {
+      //   .mockImplementationOnce(() => bagMockData.mockState)
+      //   .mockImplementationOnce(() => ({
+      //     ...bagMockData.mockState,
+      //     entities: {
+      //       ...bagMockData.mockState.entities,
+      //       bagItems: {
+      //         111111: { teste: 12345 },
+      //         [bagMockData.mockBagItem().id]: bagMockData
+      //           .mockBagItem()
+      //           .getItem({ size: bagMockData.mockSizes[1] }),
+      //       },
+      //     },
+      //   }));
+
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const bagMiddleware = require('../bag').default;
+
+      const simplifiedStore = mockSimplifiedStore(bagMockData.mockState, [
+        bagMiddleware(analytics),
+      ]);
+
+      await simplifiedStore.dispatch({
         type: bagActionTypes.UPDATE_BAG_ITEM_SUCCESS,
         payload: {
-          result: { id: mockBagId },
+          result: { id: bagMockData.mockBagId },
           entities: {
             bagItems: {
-              [mockBagItemId]: {
-                id: mockBagItemId,
-                product: mockBagProductId,
-                size: { id: sizes[0].id },
-              },
+              [bagMockData.mockBagItem().id]: bagMockData
+                .mockBagItem()
+                .getItem({ size: bagMockData.mockSizes[1] }),
             },
           },
-          bagItemId: mockBagItemId,
+          bagItemId: bagMockData.mockBagItemId,
         },
         meta: {
-          productId: mockBagProductId,
-          oldQuantity: 1,
+          bagItemId: bagMockData.mockBagItemId,
+          productId: bagMockData.mockProductId,
           quantity: 3,
-          size: sizes[0].id,
-          oldSize: sizes[1],
+          size: (sizes as SizeAdapted[])[1]?.id,
         },
       });
 
       const baseData = {
         brand: brandName,
-        cartId: mockBagId,
+        cartId: bagMockData.mockBagId,
         category: categoryName,
         currency: currencyCode,
         discountValue: discount,
-        id: mockBagProductId,
+        id: bagMockData.mockProductId,
         name: productDescription,
-        oldQuantity: 1,
+        oldQuantity: 3,
         price: priceWithDiscount,
         priceWithoutDiscount,
         quantity: 3,
         sku,
-        size: sizes[0].name,
+        size: (sizes as SizeAdapted[])[1]?.name,
         variant: colorName,
-        oldSize: sizes[1].name,
+        oldSize: (sizes as SizeAdapted[])[0]?.name,
       };
 
       expect(trackSpy).nthCalledWith(1, eventTypes.PRODUCT_UPDATED, baseData);
 
       expect(trackSpy).nthCalledWith(2, eventTypes.PRODUCT_REMOVED_FROM_CART, {
         ...baseData,
-        quantity: 1,
-        size: sizes[1].name,
+        quantity: 3,
+        size: (sizes as SizeAdapted[])[0]?.name,
       });
 
       expect(trackSpy).nthCalledWith(3, eventTypes.PRODUCT_ADDED_TO_CART, {
         ...baseData,
-        oldQuantity: 0,
+        oldQuantity: undefined,
```

Comment:
This can be removed if the value is now `undefined`

Reply:
My mistake, this is necessary to override the value of `oldQuantity` in baseData with undefined.